core/src/main/java/hudson/model/Queue.java

```diff
@@ -430,45 +452,58 @@ public synchronized void load() {
             }
         } catch (IOException e) {
             LOGGER.log(Level.WARNING, "Failed to load the queue file " + getXMLQueueFile(), e);
+        } finally { updateSnapshot(); } } finally {
```

Comment:
Java 7 would be handy here, I guess: could define a local `AutoCloseable` _[sic]_ encapsulating both the lock and the snapshot, and invoke it using `try`-with-resources.